File: ROS2/C2_3/src/python_src/demo_python_pkg/demo_python_pkg/person_node.py
import rclpy
from rclpy.node import Node


class PersonNode(Node):
    def __init__(self, name_value:str, age_value: int,node_name:str) -> None:
        super().__init__(node_name)
        self.name = name_value
        self.age = age_value

    def eat(self, food:str) -> None:    
        self.get_logger().info(f"{self.name} is eating {food}")


def main():
    rclpy.init()
    person = PersonNode("Zhangsan", 18,"person_node_san")
    person.eat("yuxiangshousi")

    rclpy.spin(person)
    rclpy.shutdown()

File: ROS2/C2_3/src/demo_cpp/src/person_node.cpp
#include "rclcpp/rclcpp.hpp"


class PersonNode : public rclcpp::Node
{
private:
    std::string name;
    int age;

public:
    PersonNode(const std::string &node_name, const std::string &name, const int &age) : Node(node_name), age(age)
    {
        this->name = name;
        this->age = age;
    }

    void eat(const std::string &food)
    {
        RCLCPP_INFO(this->get_logger(), "%s ,%d  is eating %s", this->name.c_str(),
        this->age ,food.c_str());
    }

};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<PersonNode>("ros2_cpp_node", "Frank", 25);
    node->eat("DOUFU");
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}